src/shogun/base/SGObject.cpp

```diff
@@ -1012,3 +1012,40 @@ CSGObject* CSGObject::create_empty() const
 	SG_REF(object);
 	return object;
 }
+
+namespace shogun
+{
+#define SGOBJECT_PUT_DEFINE(T) \
+void CSGObject::put(const std::string& name, T const & value)  throw(ShogunException)\
+{ \
+	Tag<T> tag(name); \
+	put(tag, value); \
+}
+
+SGOBJECT_PUT_DEFINE(SGVector<int32_t>)
+SGOBJECT_PUT_DEFINE(SGVector<float64_t>)
+SGOBJECT_PUT_DEFINE(CSGObject*)
+
+#define PUT_DEFINE_CHECK_AND_CAST(T) \
+else if (has(Tag<T>(name))) \
+	put(Tag<T>(name), (T)value);
+
+#define SGOBJECT_PUT_DEFINE_NUMBER(numeric_t) \
+void CSGObject::put(const std::string& name, numeric_t const & value)  throw(ShogunException)\
+{ \
+	/* use correct type of possible, otherwise cast-convert */ \
+	if (has(Tag<numeric_t>(name))) \
+		put(Tag<numeric_t>(name), value); \
+	PUT_DEFINE_CHECK_AND_CAST(int32_t) \
+	PUT_DEFINE_CHECK_AND_CAST(float32_t) \
```

Comment:
@lisitsyn since it is only octave that cannot decide between float and int, maybe some of this could be done in swig....though I don't know how that would work....

Reply:
https://www.gnu.org/software/octave/doc/v4.2.0/Integer-Data-Types.html